apps/site/lib/site_web/controllers/route_controller.ex

```diff
@@ -6,18 +6,42 @@ defmodule SiteWeb.RouteController do
   alias Leaflet.MapData.Polyline
   alias Routes.{Repo, Route}
 
+  @display_fare_classes [
+    :local_bus_fare,
+    :express_bus_fare,
+    :rapid_transit_fare,
+    :commuter_rail_fare,
+    :ferry_fare
+  ]
+
   @spec get_by_stop_id(Plug.Conn.t(), map) :: Plug.Conn.t()
   def get_by_stop_id(conn, %{"stop_id" => stop_id} = _params) do
+    routes = stop_id |> Repo.by_stop()
+
     routesWithPolylines =
-      stop_id
-      |> Repo.by_stop()
+      routes
       |> Enum.map(fn route ->
         route
         |> Route.to_json_safe()
         |> Map.put(:polylines, route_polylines(route, stop_id))
       end)
 
-    json(conn, routesWithPolylines)
+    fares =
+      stop_id
+      # Get fare_class for all routes at this stop and at connecting stops
+      |> Repo.by_stop(include: "stop.connecting_stops")
+      |> Enum.map(&display_fare_class/1)
```

Comment:
This function confuses me.  Don't all the routes already have a `fare_class` property?  Why does the mode need to be converted into a fare class atom?  Also doesn't this defeat the purpose of using the `fare_class` property on a route in the first place?

Reply:
Maybe I could name it something better; the intent was to translate the actual fare class into the one we ultimately wanted to display on the page. See comment on `display_fare_class/1` function.
It's basically compensating for the fact that we don't have any thought out logic or design for displaying special fares or free fares right now - if we remedy this later on, we could deprecate this function.
Hope that makes some sense.